r/tools/nixlibs.R

```diff
@@ -860,7 +860,8 @@ if (is_release) {
   VERSION <- VERSION[1, 1:3]
   arrow_repo <- paste0(getOption("arrow.repo", sprintf("https://apache.jfrog.io/artifactory/arrow/r/%s", VERSION)), "/libarrow/")
 } else {
-  not_cran <- TRUE
+  # Don't override explictily set NOT_CRAN env var, as it is used in CI.
+  not_cran <- TRUE && !env_is("NOT_CRAN", "false")
```

Comment:
```suggestion
  not_cran <- !env_is("NOT_CRAN", "false")
```

Reply:
The comment of this should be updated too?

Reply:
> The comment of this should be updated too?

I don't understand? I already added a comment?

Reply:
Ah, we have `not_cran <- env_is("NOT_CRAN", "true")` in the above line.

We may be able to simplify this like the following:

```diff
diff --git a/r/tools/nixlibs.R b/r/tools/nixlibs.R
index fe8de284b1..e89b814512 100644
--- a/r/tools/nixlibs.R
+++ b/r/tools/nixlibs.R
@@ -854,13 +854,12 @@ on_windows <- tolower(Sys.info()[["sysname"]]) == "windows"
 # For local debugging, set ARROW_R_DEV=TRUE to make this script print more
 quietly <- !env_is("ARROW_R_DEV", "true")
 
-not_cran <- env_is("NOT_CRAN", "true")
+not_cran <- !env_is("NOT_CRAN", "false")
 
 if (is_release) {
   VERSION <- VERSION[1, 1:3]
   arrow_repo <- paste0(getOption("arrow.repo", sprintf("https://apache.jfrog.io/artifactory/arrow/r/%s", VERSION)), "/libarrow/")
 } else {
-  not_cran <- TRUE
   arrow_repo <- paste0(getOption("arrow.dev_repo", "https://nightlies.apache.org/arrow/r"), "/libarrow/")
 }
 
```

Reply:
No, this is intentional. We only want to default to setting `not_cran <- TRUE` if we are on a dev version. For release versions people have to explicitly opt in, this is needed for CRAN (as the name implies^^).

Reply:
I see!